tests/FINDOLOGIC/Tests/DataElementsTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace FINDOLOGIC\Tests;
+
+use FINDOLOGIC\Export\Data\Attribute;
+use FINDOLOGIC\Export\Data\Bonus;
+use FINDOLOGIC\Export\Data\DateAdded;
+use FINDOLOGIC\Export\Data\Description;
+use FINDOLOGIC\Export\Data\Keyword;
+use FINDOLOGIC\Export\Data\Name;
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\Export\Data\Price;
+use FINDOLOGIC\Export\Data\Property;
+use FINDOLOGIC\Export\Data\SalesFrequency;
+use FINDOLOGIC\Export\Data\Sort;
+use FINDOLOGIC\Export\Data\Summary;
+use FINDOLOGIC\Export\Data\Url;
+use FINDOLOGIC\Export\Helpers\EmptyValueNotAllowedException;
+use PHPUnit\Framework\TestCase;
+
+class DataElementsTest extends TestCase
+{
+    public function multiValueItemProvider()
+    {
+        return array(
+            'Keyword with empty value' => array('', Keyword::class, true),
+            'Keyword with value' => array('value', Keyword::class, false),
+            'Ordernumber with empty value' => array('', Ordernumber::class, true),
+            'Ordernumber with value' => array('value', Ordernumber::class, false)
+        );
+    }
+
+    /**
+     * @dataProvider multiValueItemProvider
+     */
+    public function testAddingEmptyValuesToMultiValueItemCausesException($value = '', $elementtype, $shouldCauseException = true)
+    {
+        try {
+            $element = new $elementtype($value);
+            if ($shouldCauseException) {
+                $this->fail('Adding empty values should cause exception!');
+            }
+        } catch (\Exception $exception) {
+            $this->assertEquals(EmptyValueNotAllowedException::class, get_class($exception));
+        }
+    }
+
+    public function simpleValueItemProvider()
```

Comment:
Add a doc block describing the structure that each data set has.

Reply:
Done

Reply:
What I meant was listing a description of which values is being provided. In this case, something like element value, element class, and whether that value for this particular class should cause an exception. In other words: What's the structure of this mysterious array?

Reply:
Done